src/server/req_delete.c

```diff
@@ -201,103 +201,169 @@ acct_del_write(char *jid, job *pjob, struct batch_request *preq, int nomail)
  *		done by req_deletejob()
  *
  * @param[in]	preq	- Batch request structure.
+ * @param[in]	pjob	- pointer to job
+ * @param[in]	parent	- pointer to parent job for sub job
+ * @param[in]	jt	- job type
  *
  * @return	int
  * @retval	TRUE	- Job history  has been purged
  * @retval	FALSE	- Job is not a history job
  */
 int
-check_deletehistoryjob(struct batch_request * preq)
+check_deletehistoryjob(struct batch_request * preq, job *pjob, job *parent, int jt)
```

Comment:
The prototype of this function is now changed to return pointer to job and its parent in case of subjob. This information can easily be derived from find_job() and find_arrayparent() in the caller itself. Moreover check_deletehistoryjob() is getting called only from req_deletejob only. Is there any specific reason why you changed the prototype ?

Reply:
Actually this function does not return the job pointer. Here the pjob and parent job pointer are in coming parameters.
Justification: since the caller already has the pjob and parent, its quicker to pass them instead of doing the search all over again.

Reply:
As discussed check_deletehistoryjob() is getting called from req_deletejob() which is getting called in dispatch_request() whenever a request comes from the client like qdel for example.  Assume in future there is a need to call this function directly from dispatch_request() then the current prototype supports it easily. But since you changed the prototype if it has to be called from dispatch_requestt() we have to fill up the additional parameters to call it. I hope this was the reason why the signature of the functiton check_deletehistoryjob() was designed initially as follows.
int check_deletehistoryjob(struct batch_request * preq)

I suggest not to change the prototype. If you change the prototype now and there is need to call this function directly from dispatch_request() then again we have to revisit the code and change it accordingly. Instead if we maintain the same prototype it is easy for future enhancements.

Reply:
>Assume in future there is a need to call this function directly from dispatch_request() ..... I suggest not to change the prototype. If you change the prototype now and there is need to call this function directly from dispatch_request() then again we have to revisit the code and change it accordingly. Instead if we maintain the same prototype it is easy for future enhancements.

check_deletehistoryjob() is just a local function limited to req_delete.c and not exposed as an API so I feel we can safely add new parameters to help speed up things. Further I feel the assumption made is a bit unwarranted to have any considerations here. Let me know if you still feel the parameters are unnecessary and there is something to justify the assumptions made on the future changes.